src/components/Collapsible/Collapsible.tsx

```diff
@@ -30,165 +24,68 @@ export interface CollapsibleProps {
 
-interface State {
-  height?: number | null;
-  animationState: AnimationState;
-  open: boolean;
-}
-
-const ParentCollapsibleExpandingContext = createContext(false);
-
-class CollapsibleInner extends Component<CollapsibleProps, State> {
-  static contextType = ParentCollapsibleExpandingContext;
-
-  static getDerivedStateFromProps(
-    {open: willOpen}: CollapsibleProps,
-    {open, animationState: prevAnimationState}: State,
-  ) {
-    let nextAnimationState = prevAnimationState;
-    if (open !== willOpen) {
-      nextAnimationState = 'measuring';
-    }
-
-    return {
-      animationState: nextAnimationState,
-      open: willOpen,
-    };
-  }
-
-  context!: React.ContextType<typeof ParentCollapsibleExpandingContext>;
-
-  state: State = {
-    height: null,
-    animationState: 'idle',
-    // eslint-disable-next-line react/no-unused-state
-    open: this.props.open,
+export function Collapsible({
+  id,
+  expandOnPrint,
+  open,
+  transition,
+  children,
+}: CollapsibleProps) {
+  const [height, setHeight] = useState<Number | null>(null);
+  const [isOpen, setIsOpen] = useState(open);
+  const collapisbleContainer = useRef(null);
+
+  const wrapperClassName = classNames(
+    styles.Collapsible,
+    expandOnPrint && styles.expandOnPrint,
+    isOpen && styles.open,
+    height && styles.animating,
+  );
+
+  const collapsibleStyles = {
+    ...(transition && {
+      transitionDuration: `${transition.duration}`,
+      transitionTimingFunction: `${transition.timingFunction}`,
+    }),
+    ...(typeof height === 'number' && {
+      height: `${height}px`,
+      overflow: 'hidden',
+    }),
   };
 
-  private node = createRef<HTMLDivElement>();
-  private heightNode = createRef<HTMLDivElement>();
-
-  componentDidUpdate({open: wasOpen}: CollapsibleProps) {
-    const {animationState} = this.state;
-    const parentCollapsibleExpanding = this.context;
-
-    if (parentCollapsibleExpanding && animationState !== 'idle') {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({
-        animationState: 'idle',
-      });
-
+  useEffect(() => {
+    if (open === isOpen) {
       return;
     }
 
-    requestAnimationFrame(() => {
-      const heightNode = this.heightNode.current;
-      switch (animationState) {
-        case 'idle':
-          break;
-        case 'measuring':
-          this.setState({
-            animationState: wasOpen ? 'closingStart' : 'openingStart',
-            height: wasOpen && heightNode ? heightNode.scrollHeight : 0,
-          });
-          break;
-        case 'closingStart':
-          this.setState({
-            animationState: 'closing',
-            height: 0,
-          });
-          break;
-        case 'openingStart':
-          this.setState({
-            animationState: 'opening',
-            height: heightNode ? heightNode.scrollHeight : 0,
-          });
-      }
-    });
-  }
+    setHeight(collapisbleContainer.current.scrollHeight);
+  }, [open]);
```

Comment:
This useEffect measured the component and applies the height. For open this is the only effect ran as it measures and applies, this causes the transition from 0 -> measured height. For closing we do the same thing but then use the second useEffect to transition to 0.